tests/edit/test_main.py

```diff
@@ -558,3 +558,64 @@ def test_required_fields_red_asterisk(
             expect(asterisk).to_have_css("color", "rgb(255, 0, 0)")
         else:
             expect(asterisk).to_have_count(0)
+
+
+@pytest.mark.integration
+def test_edit_page_warn_tab_close(edit_page: Page) -> None:
+    page = edit_page
+
+    # ensure navigation without changes is working
+    page.goto("https://www.rki.de")
+    page.screenshot(
+        path="tests_edit_test_main-test_edit_page_warn_tab_close-rki_website.png"
+    )
+    assert "https://www.rki.de" in page.url
+
+    # back to edit site
+    page.go_back()
+    page.wait_for_url("**/item/**")
+    page.wait_for_selector(
+        "[data-testid='new-additive-alternativeTitle-00000000000000']"
+    )
+    page.screenshot(
+        path="tests_edit_test_main-test_edit_page_warn_tab_close-back_on_edit.png"
+    )
+
+    # change a value
+    page.get_by_test_id("new-additive-alternativeTitle-00000000000000").click()
+    page.get_by_test_id("additive-rule-alternativeTitle-0-text").fill(
+        "new alternative title"
+    )
+
+    handle_dialog_called: list[bool] = []
+
+    def _handle_dialog(dialog: Dialog) -> None:
+        # damn i love python
```

Comment:
😬

you could pass the reference to the list along with the handle function to the page.on(dialog, ...) if you wanted to avoid this. i'd find that a bit more readable:

```python
def _handle_dialog(was_called: list[bool], dialog: Dialog) -> None:
   ...

from functools import partial

page.on("dialog", partial(_handle_dialog, handle_dialog_called))
```

Reply:
fine, since it's just in a test, let's have at it 😄